src/task_list/im_rad_task_list.hpp

```diff
@@ -0,0 +1,170 @@
+#ifndef TASK_LIST_IM_RAD_TASK_LIST_HPP_
+#define TASK_LIST_IM_RAD_TASK_LIST_HPP_
+//========================================================================================
+// Athena++ astrophysical MHD code
+// Copyright(C) 2014 James M. Stone <jmstone@princeton.edu> and other code contributors
+// Licensed under the 3-clause BSD License, see LICENSE file for details
+//========================================================================================
+//! \file im_rad_task_list.hpp
+//! \brief define im_rad_task_list class
+//! \brief This is used to handle the boundary condition
+
+// C headers
+
+// C++ headers
+
+// Athena++ headers
+#include "../athena.hpp"
+#include "./task_list.hpp"
+
+// forward declarations
+class Mesh;
+class MeshBlock;
+class Radiation;
+class IMRadTaskList;
+class TaskID;
+
+
+//----------------------------------------------------------------------------------------
+//! \struct IMRadTask
+//! \brief data and function pointer for an individual IMRadTask
+
+struct IMRadTask {
+  TaskID task_id;      //!> encodes task using bit positions in IMRadTaskNames
+  TaskID dependency;   //!> encodes dependencies to other tasks using IMRadTaskNames
+  TaskStatus (IMRadTaskList::*TaskFunc)(MeshBlock *); //!> ptr to a task
+};
+
+//----------------------------------------------------------------------------------------
+//! \class MultigridTaskList
+//! \brief data and function definitions for IMRadTaskList class
+
+class IMRadTaskList {
```

Comment:
I just noticed that `MultigridTaskList` is not derived either?

Reply:
True, but it is intentional for a few reasons. MG works on Multigrid objects, not MeshBlocks. This is for future extension to apply different parallelizations for MG and the main part. Maybe I can still derive it from the TaskList, though.

For this radiation, I think it can be done similarly to STS Task List.

Reply:
I think the `IMRadTaskList` has a similar argument as the `MultigridTaskList` then, since the function signature is also different. It does not have a notion of a time integrator stage:
```
TaskStatus (IMRadTaskList::*TaskFunc)(MeshBlock *); //!> ptr to a task
```

The `SuperTimeStepTaskList` does use `int stage` in a similar fashion to the main task list, so it isnt quite the same as the radiation task list.

Anyway, let's continue this discussion on #537